### PR TITLE
adding decimal point and retraction for keyboard

### PR DESCRIPTION
**Pull request by zhbh**
This branch has solved two issues as follow:
1. Keyboard in unbond DHX screen does not go away once you bring it up #266
2. Keyboard in unbond DHX screen needs a decimal point #265

### CODE REVIEW

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart [hunk 1]
```diff
@@ -89,115 +89,133 @@ class _DhxBondingPageState extends State<DhxBondingPage> {
...
-                          .read<SupernodeDhxCubit>()
-                          .confirmBondUnbond(bond: ctrl.text.trim()),
-                      buttonTitle: FlutterI18n.translate(context, 'confirm'),
-                      bgColor: Token.supernodeDhx.color),
-                ],
-              ),
-            ),
-          ]),
+                ),
+              ])),
```
Comment (festelo) [inline lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart:218]: commas

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_unbonding_page.dart [hunk 2]
```diff
@@ -86,80 +86,89 @@ class _DhxUnbondingPageState extends State<DhxUnbondingPage> {
...
-                      child: Image.asset(
-                        AppImages.iconUnbond,
-                        color: Colors.white,
-                      ),
-                    ),
-                    SizedBox(width: 20),
-                    Flexible(
-                      child: Column(
+      child: GestureDetector(
+          onTap: () => FocusScope.of(context).requestFocus(FocusNode()),
```
Comment (festelo) [inline lib/page/home_page/wallet/supernode_dhx_token/dhx_unbonding_page.dart:90]: see the comment above about focusNode

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_unbonding_page.dart [hunk 3]
```diff
@@ -86,80 +86,89 @@ class _DhxUnbondingPageState extends State<DhxUnbondingPage> {
...
-                          .read<SupernodeDhxCubit>()
-                          .confirmBondUnbond(unbond: ctrl.text.trim()),
-                      buttonTitle: FlutterI18n.translate(context, 'confirm'),
-                      bgColor: Token.supernodeDhx.color),
-                ],
-              ),
-            ),
-          ]),
+                ),
+              ])),
```
Comment (festelo) [inline lib/page/home_page/wallet/supernode_dhx_token/dhx_unbonding_page.dart:171]: commas

**Review by festelo — CHANGES_REQUESTED** [review]

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart [hunk 4]
```diff
@@ -89,115 +89,133 @@ class _DhxBondingPageState extends State<DhxBondingPage> {
...
-                      child: Image.asset(
-                        AppImages.iconBond,
-                        color: Colors.white,
-                      ),
-                    ),
-                    SizedBox(width: 20),
-                    Flexible(
-                      child: Column(
+      child: GestureDetector(
+          onTap: () => FocusScope.of(context).requestFocus(FocusNode()),
```
[the remark this answers is not in the document]
Comment (zhbh) [reply lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart:93]: Okay, Thanks.

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart [hunk 5]
```diff
@@ -89,115 +89,133 @@ class _DhxBondingPageState extends State<DhxBondingPage> {
...
+                                          TextSpan(
+                                              text: FlutterI18n.translate(
+                                                  context,
+                                                  'bond_dhx_instruction_1')),
+                                          TextSpan(
+                                              text: FlutterI18n.translate(
+                                                  context, 'click_here'),
+                                              style: kMiddleFontOfBlueLink,
+                                              recognizer:
+                                                  new TapGestureRecognizer()
```
[the remark this answers is not in the document]
Comment (zhbh) [reply lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart:148]: Someone uses, I will delete that.

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart [hunk 6]
```diff
@@ -89,115 +89,133 @@ class _DhxBondingPageState extends State<DhxBondingPage> {
...
+                                                    ..onTap = () =>
+                                                        openSupernodeDeposit(
+                                                            context,
+                                                            Token
+                                                                .supernodeDhx)),
```
[the remark this answers is not in the document]
Comment (zhbh) [reply lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart:153]: What?

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhbh) [reply lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart:93]: I search requestFocus in the project, which many places use likes that. And I replace all with unfocus().

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (zhbh) [reply lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart:153]: IDE can add comma automatically at the end of statement. So Flutter format can't make it well.

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart [hunk 7]
```diff
@@ -89,115 +89,133 @@ class _DhxBondingPageState extends State<DhxBondingPage> {
...
+                                                        Navigator.pushNamed(
+                                                            context,
+                                                            'lock_page',
+                                                            arguments: {
+                                                              'isDemo': context
+                                                                  .read<
+                                                                      AppCubit>()
+                                                                  .state
+                                                                  .isDemo
+                                                            })),
```
[the remark this answers is not in the document]
Comment (zhbh) [reply lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart:181]: delete commas

File: lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart (hunk 6 above)
[the remark this answers is not in the document]
Comment (zhbh) [reply lib/page/home_page/wallet/supernode_dhx_token/dhx_bonding_page.dart]: delete commas

**Pull request merged**